source/material_model/rheology/visco_plastic.cc

```diff
@@ -874,14 +865,11 @@ namespace aspect
             // average over the volume volume fractions
             for (unsigned int j = 0; j < volume_fractions.size(); ++j)
               {
-                plastic_out->cohesions[i] += volume_fractions[j] * cohesions[j];
-                // Also convert radians to degrees
-                plastic_out->friction_angles[i] += constants::radians_to_degree * volume_fractions[j] * friction_angles_RAD[j];
+                const Rheology::DruckerPragerParameters &drucker_prager_parameters = isostrain_viscosities.drucker_prager_parameters[j];
```

Comment:
Nice! This way, any new parameter added to `drucker_prager_parameters` will not require manually handling the phase averaging here (or any other operation).